### PR TITLE
`vms/platformvm`: Change `AdvanceTimeTo` to modify passed-in `parentState`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Whenever we consume `AdvanceTimeTo`, we always apply the changes immediately. Let's move that into this function directly.

## How this works

pretty straightforward

## How this was tested

CI

### CODE REVIEW

**Review by abi87 — APPROVED** [review]
lgtm. I think given the introducing of the `NewDiffOn` we can avoid the `stateChanges` struct

File: vms/platformvm/block/builder/builder.go [hunk 1]
```diff
@@ -319,12 +319,10 @@ func packBlockTxs(
 		return nil, err
 	}
 
-	changes, err := txexecutor.AdvanceTimeTo(backend, stateDiff, timestamp)
+	_, err = txexecutor.AdvanceTimeTo(backend, stateDiff, timestamp)
 	if err != nil {
 		return nil, err
 	}
```
Comment (danlaine) [inline vms/platformvm/block/builder/builder.go:325]: nit; I like to keep the scope smaller when possible
```suggestion
	if _, err := txexecutor.AdvanceTimeTo(backend, stateDiff, timestamp); if err != nil {
		return nil, err
	}
```

File: vms/platformvm/block/executor/verifier.go [hunk 2]
```diff
@@ -65,20 +65,16 @@ func (v *verifier) BanffProposalBlock(b *block.BanffProposalBlock) error {
 		return err
 	}
 
-	// Apply the changes, if any, from advancing the chain time.
-	nextChainTime := b.Timestamp()
-	changes, err := executor.AdvanceTimeTo(
+	// Advance the time to [b.Timestamp()].
+	_, err = executor.AdvanceTimeTo(
```
Comment (danlaine) [inline vms/platformvm/block/executor/verifier.go:69]: same nit as above about reducing scope
```suggestion
	if _, err := executor.AdvanceTimeTo(
```

File: vms/platformvm/txs/executor/proposal_tx_executor.go [hunk 3]
```diff
@@ -296,15 +296,11 @@ func (e *ProposalTxExecutor) AdvanceTimeTx(tx *txs.AdvanceTimeTx) error {
 		return err
 	}
 
-	changes, err := AdvanceTimeTo(e.Backend, e.OnCommitState, newChainTime)
+	_, err = AdvanceTimeTo(e.Backend, e.OnCommitState, newChainTime)
```
Comment (danlaine) [inline vms/platformvm/txs/executor/proposal_tx_executor.go:299]: same nit about scope

File: vms/platformvm/txs/executor/tx_mempool_verifier.go [hunk 4]
```diff
@@ -114,12 +114,10 @@ func (v *MempoolTxVerifier) standardBaseState() (state.Diff, error) {
 		return nil, err
 	}
 
-	changes, err := AdvanceTimeTo(v.Backend, state, nextBlkTime)
+	_, err = AdvanceTimeTo(v.Backend, state, nextBlkTime)
```
Comment (danlaine) [inline vms/platformvm/txs/executor/tx_mempool_verifier.go:117]: same nit about scope

File: vms/platformvm/txs/executor/state_changes.go [hunk 5]
```diff
@@ -57,78 +57,35 @@ func VerifyNewChainTime(
...
-	return len(s.currentValidatorsToAdd) + len(s.currentDelegatorsToAdd) +
-		len(s.pendingValidatorsToRemove) + len(s.pendingDelegatorsToRemove) +
-		len(s.currentValidatorsToRemove)
-}
-
-// AdvanceTimeTo does not modify [parentState].
-// Instead it returns all the StateChanges caused by advancing the chain time to
-// the [newChainTime].
+// AdvanceTimeTo applies all state changes to [parentState] resulting from
+// advancing the chain time to [newChainTime].
```
Comment (danlaine) [inline vms/platformvm/txs/executor/state_changes.go:61]:
```suggestion
// advancing the chain time to [newChainTime].
// Returns true iff the validator set changed.
```

**Review by danlaine — APPROVED** [review]
Just some nits. Also, worth considering renaming this file from `state_changes.go` to `advance_time.go` or something.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**